src/helpers/helpers.options.ts

```diff
@@ -62,10 +64,10 @@ export function _readValueToProps(value, props) {
 
 /**
  * Converts the given value into a TRBL object.
- * @param {number|object} value - If a number, set the value to all TRBL component,
+ * @param value - If a number, set the value to all TRBL component,
  *  else, if an object, use defined properties and sets undefined ones to 0.
  *  x / y are shorthands for same value for left/right and top/bottom.
- * @returns {object} The padding values (top, right, bottom, left)
+ * @returns The padding values (top, right, bottom, left)
  * @since 3.0.0
  */
 export function toTRBL(value) {
```

Comment:
@dangreen the `_readValueToProps` new type wont work for this function, maybe it will need a little refactor? im a little confused

Reply:
@luckened What exactly doesn't work?

Reply:
![image](https://user-images.githubusercontent.com/42002892/196178724-262a41fd-1499-4044-8087-db32e22a99cf.png)
🤔

Reply:
@luckened You also should provide overload for object argument. My suggestion is only for array.

Reply:
@dangreen take another look, please

is this broken ci step maybe a flaky test? couldn't reproduce it locally

Reply:
```ts
export function _readValueToProps<K extends string>(value: number | Record<K, number>, props: K[]): Record<K, number>;
export function _readValueToProps<K extends string, T extends string>(value: number | Record<K & T, number>, props: Record<T, K>): Record<T, number>;
export function _readValueToProps(value: number | Record<string, number>, props: string[] | Record<string, string>) {
  const ret = {};
  const objProps = isObject(props);
  const keys = objProps ? Object.keys(props) : props;
  const read = isObject(value)
    ? objProps
      ? prop => valueOrDefault(value[prop], value[props[prop]])
      : prop => value[prop]
    : () => value;

  for (const prop of keys) {
    ret[prop] = numberOrZero(read(prop));
  }
  return ret;
}

/**
 * Converts the given value into a TRBL object.
 * @param value - If a number, set the value to all TRBL component,
 *  else, if an object, use defined properties and sets undefined ones to 0.
 *  x / y are shorthands for same value for left/right and top/bottom.
 * @returns The padding values (top, right, bottom, left)
 * @since 3.0.0
 */
export function toTRBL(value: number | TRBL | Point) {
  return _readValueToProps(value, {top: 'y', right: 'x', bottom: 'y', left: 'x'});
}

/**
 * Converts the given value into a TRBL corners object (similar with css border-radius).
 * @param value - If a number, set the value to all TRBL corner components,
 *  else, if an object, use defined properties and sets undefined ones to 0.
 * @returns The TRBL corner values (topLeft, topRight, bottomLeft, bottomRight)
 * @since 3.0.0
 */
export function toTRBLCorners(value: number | TRBLCorners) {
  return _readValueToProps(value, ['topLeft', 'topRight', 'bottomLeft', 'bottomRight']);
}

/**
 * Converts the given value into a padding object with pre-computed width/height.
 * @param value - If a number, set the value to all TRBL component,
 *  else, if an object, use defined properties and sets undefined ones to 0.
 *  x / y are shorthands for same value for left/right and top/bottom.
 * @returns The padding values (top, right, bottom, left, width, height)
 * @since 2.7.0
 */
export function toPadding(value?: number | TRBL): ChartArea {
  const obj: ChartArea = {
    ...toTRBL(value),
    width: 0,
    height: 0,
    top: 0,
    left: 0,
    right: 0,
    bottom: 0
  };

  obj.width = obj.left + obj.right;
  obj.height = obj.top + obj.bottom;

  return obj;
}
```